digits/tools/create_db.py

```diff
@@ -693,7 +725,10 @@ def _save_means(image_sum, image_count, mean_files):
             with open(mean_file, 'wb') as outfile:
                 outfile.write(blob.SerializeToString())
         elif mean_file.lower().endswith(('.jpg', '.jpeg', '.png')):
-            image = PIL.Image.fromarray(mean)
+            if np.issubdtype(mean.dtype, np.float):
+                image = PIL.Image.fromarray(mean*255/mean.max()).convert('L')
```

Comment:
@gheinrich , it looks like if the max value is greater than 255, but much smaller than 65536 (for example, around 800), and we save it to png file.  That image is almost black.  The reason, I believe, is the dynamic range of 16-bit PNG is much higher than mean pixel value.  We can either scale it up to full dynamic range of 16-bit PNG, or scale it down to 8-bit png/jpg.  Both look the same on screen.